libraries/chain/dbs_account.cpp

```diff
@@ -61,21 +61,15 @@ const account_object& dbs_account::create_initial_account(const account_name_typ
     FC_ASSERT(new_account_name.size() > 0, "Account 'name' should not be empty.");
     FC_ASSERT(balance.symbol == SCORUM_SYMBOL, "Invalid asset type (symbol) for balance.");
 
-    const auto& new_account = db_impl().create<account_object>([&](account_object& acc) {
-        acc.name = new_account_name;
-        acc.memo_key = memo_key;
-        acc.balance = balance;
-        fc::from_string(acc.json_metadata, json_metadata);
-        acc.recovery_account = recovery_account;
-    });
+    authority owner;
+    owner.add_authority(memo_key, 1);
+    owner.weight_threshold = 1;
+    authority active = owner;
+    authority posting = active;
+    const auto& new_account
+        = _create_account_objects(new_account_name, recovery_account, memo_key, json_metadata, owner, active, posting);
```

Comment:
Why just not pass owner variable to _create_account_objects() as active and posting parameters?